src/main/java/com/ably/kafka/connect/ChannelSinkTask.java

```diff
@@ -123,9 +124,12 @@ public void put(Collection<SinkRecord> records) {
 
                 channel.publish(message);
             } catch (AblyException e) {
-                // The ably client should attempt retries itself, so if we do have to handle an exception here,
-                // we can assume that it is not retryably.
-                throw new ConnectException("ably client failed to publish", e);
+                if(ably.options.queueMessages){
+                    logger.error("Failed to publish message", e);
+                }else{
+                    throw new RetriableException(e.getMessage());
```

Comment:
I can't see any reason you should not pass on the cause. e.g.:

```suggestion
                    throw new RetriableException("Failed to publish to Ably when queueMessages is disabled.", e);
```

Reply:
https://github.com/ably/kafka-connect-ably/pull/36/commits/027bf56b0e6c11c8d33ca9f47e3d6c9d186ac407